craft_parts/executor/step_handler.py

```diff
@@ -108,6 +109,26 @@ def run_builtin(self) -> StepContents:
     def _builtin_pull(self) -> StepContents:
         if self._source_handler:
             self._source_handler.pull()
+
+        # Plugin prepare commands.
+        if self._step_info.offline_build:
```

Comment:
This does not make much sense to me, if the plugin is strict and we are offline, how will we pull?

Reply:
`offline_build` means that the build step will work on an offline environment, the pull step still requires network access to download content.